src/server/worker/api_server.go

```diff
@@ -551,8 +552,14 @@ func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLog
 			}
 		}
 	} else {
-		if err := os.MkdirAll(outPath, 0777); err != nil {
-			return "", err
+		if a.pipelineInfo.S3Out {
+			if err := os.RemoveAll(outPath); err != nil {
+				return "", fmt.Errorf("couldn't remove %q for s3-out pipeline: %v", outPath, err)
+			}
```

Comment:
I'm assuming this is for `/pfs/out` -- if so, why the removal? Shouldn't it never be created in the first place?

Reply:
This is a mistake—I was coding a bit defensively to try and get the PR out. In particular, I wasn't sure whether the `out` directory might be left behind from a previous datum but I now see that this is in a UUID subdirectory of the scratch directoy, so no, that can't happen, and I've gotten rid of the `RemoveAll` here and in `linkData`.